tests/cost_report/infrastructure/test_multi_infrastructure_profile_costs_calculator_builder.py

```diff
@@ -0,0 +1,140 @@
+)
+from tests.conftest import PointSurroundingsTestCase
+from tests.cost_report import infrastructure
+
+
+class TestMultiInfrastructureProfileCostsCalculatorBuilder:
+    def test_initialize(self):
+        # 1. Define test data.
+        _builder = MultiInfrastructureProfileCostsCalculatorBuilder(
+            cost_settings=None, infrastructure_wrapper=None, surtax_cost_settings=None
+        )
+
+        # 2. Verify expectations
+        assert isinstance(_builder, MultiInfrastructureProfileCostsCalculatorBuilder)
+        assert isinstance(_builder, BuilderProtocol)
+        assert not _builder.cost_settings
+        assert not _builder.infrastructure_wrapper
+        assert not _builder.surtax_cost_settings
+
+    @pytest.mark.parametrize(
+        "surtax_cost_factor",
+        [
+            pytest.param(_sf, id=f"Surtax Factor: {_sf.name.lower()}")
+            for _sf in SurtaxFactorEnum
+        ],
+    )
+    @pytest.mark.parametrize(
+        "dh0_factor",
+        [
+            pytest.param(_ice, id=f"Infra costs: {_ice.name.lower()}")
+            for _ice in InfraCostsEnum
+        ],
+    )
+    def test_build_given_different_settings(
+        self,
+        surtax_cost_factor: SurtaxFactorEnum,
+        dh0_factor: InfraCostsEnum,
+        surroundings_infrastructure_fixture: tuple[
+            SurroundingsInfrastructure, PointSurroundingsTestCase
+        ],
+    ):
+        # 1. Define test data.
+        _infrastructure, _test_case_data = surroundings_infrastructure_fixture
+        _infra_wrapper = InfrastructureSurroundingsWrapper(
+            infrastructures_considered=True,
+            surtax_cost_factor=surtax_cost_factor,
+            non_rising_dike_costs_factor=dh0_factor,
+            roads_class_unknown_polderside=_infrastructure,
+        )
+        _infrastructure.infrastructure_name = "roads_class_unknown_polderside"
+
+        _adding_roads = 24.0
+        _removing_roads = 12.0
+
+        _builder = MultiInfrastructureProfileCostsCalculatorBuilder(
+            cost_settings=InfrastructureCostsSettings(
+                adding_roads_unknown=_adding_roads,
+                removing_roads_unknown=_removing_roads,
+            ),
+            infrastructure_wrapper=_infra_wrapper,
+            surtax_cost_settings=SurtaxCostsSettings(
+                roads_normal=1.0, roads_easy=0.5, roads_hard=2.0
+            ),
+        )
+
+        _expected_surtax_cost = 1.0
+        if surtax_cost_factor == SurtaxFactorEnum.MAKKELIJK:
+            _expected_surtax_cost = 0.5
+        if surtax_cost_factor == SurtaxFactorEnum.MOEILIJK:
+            _expected_surtax_cost = 2
+
+        _expected_zone_a_costs = 0
+        _expected_zone_b_costs = _adding_roads + _removing_roads
+        if dh0_factor == InfraCostsEnum.HERSTEL:
+            _expected_zone_a_costs = _adding_roads
+        if dh0_factor == InfraCostsEnum.VERVANG:
+            _expected_zone_a_costs = _adding_roads + _removing_roads
```

Comment:
Shouldn't these be separate cases?

Reply:
Yes, it should, and I would usually do so. However then I would not be able to pull the trick of crossing test cases, which in this case allows me to generate:
- All possible combinations of `SurtaxFactorEnum` (3).
- All possible combinations of `InfraCostsEnum` (3).
- All possible combinations (already used elsewhere) of basic `PointSurroundings` cases and matrices (7).
Adding up to 73 different test cases which self validate here.

If I were to do what you suggest, which again you are right, it would mean that I need to create a total of 9 different tests to represent the combinations of `SurtaxFactorEnum` and `InfraCostsEnum` which in my opinion is simply not worth it considering just a few `if-else` statements in this case do the work. If we are interested into testing some specific cases we can still do it though.

In case you are not convinced / disagree on this we can still discuss about it.

Reply:
As an extra remark, keep in mind that I'm only modifying one of the infrastructures and checking whether this is reflected in the results. Therefore you'd still need at least an `if-else` to check whether the property being validated gives a different result than the rest.